### PR TITLE
Suppress all warnings from third-party projects.

### PR DESCRIPTION
**Pull request by antiagainst**

### CODE REVIEW

File: third_party/CMakeLists.txt [hunk 1]
```diff
@@ -1,3 +1,6 @@
+# Suppress all warnings from third-party projects.
+set_property(DIRECTORY APPEND PROPERTY COMPILE_OPTIONS -w)
```
Comment (dmircevski) [inline third_party/CMakeLists.txt]: Does this work for all compilers?

Comment (antiagainst) [inline third_party/CMakeLists.txt]: Works on GCC and Clang. "[When compiling with MSVC the “-w” will be automatically converted to “/w” which has the same function.](https://www.johnlamp.net/cmake-tutorial-4-libraries-and-subdirectories.html)"

Comment (dmircevski) [inline third_party/CMakeLists.txt]: Perfect, thx!

**Comment by dmircevski** [comment]
+2

**Pull request merged**